internal/server/singleprocess/service_instance.go

```diff
@@ -18,8 +21,43 @@ func (s *service) ListInstances(
 	var err error
 	switch scope := req.Scope.(type) {
 	case *pb.ListInstancesRequest_DeploymentId:
-		result, err = s.state.InstancesByDeployment(scope.DeploymentId, nil)
+		// For more complex deployments, sometimes it takes a bit longer for
+		// the instance to respond. This blocks on a response when querying
+		// a deployment by id, and lets the user know it is taking a while
+		//
+		// The default case is no timeout, no blocking. Make a request and return
+
+		timeout := "0"
+		if req.ConnectTimeout != "" {
+			timeout = req.ConnectTimeout
+		}
+		connectTimeout, err := time.ParseDuration(timeout)
+		if err != nil {
+			return nil, err
+		}
+
+		if connectTimeout > 0 {
+			_, cancel := context.WithTimeout(ctx, connectTimeout)
```

Comment:
you gotta assign the context here, so `ctx, cancel =` otherwise the wait below won't wait on the timeout one.

Reply:
Hmm, when I had initially done that the compiler said `ctx` isn't used :thinking: I tried naming it to something else and I got the same error

Reply:
Yeah you gotta use `=` and not `:=` so that it overwrites the existing value. You may have to set `var cancel context.CancelFunc` above this line so that that is declared in that case.

Reply:
Ohhh right, my eyes skipped over the `:` :+1: thanks